### PR TITLE
Automation for Storage Pool

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
Automation for Storage Pool

### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/storage_pool.pp [hunk 1]
```diff
@@ -23,14 +23,14 @@
...
-      storageSystemUri => 'ThreePAR-2'
-      # storageSystemUri => '/rest/storage-systems/TXQ1000307'
+      storageSystemUri => 'ThreePAR-1'
+      #storageSystemUri => '/rest/storage-systems/TXQ1000307'
     }
 }
 
 oneview_storage_pool{'storage_pool_2':
     ensure  => 'found',
-    require => Oneview_storage_pool['storage_pool_1'],
```
Comment (VenkateshRavula) [inline examples/storage_pool.pp]: any reason for removing comma..?

File: spec/unit/provider/oneview_storage_pool_spec.rb [hunk 2]
```diff
@@ -19,33 +19,54 @@
...
 describe provider_class, unit: true do
   include_context 'shared context'
+  context 'given the minimum parameters' do
+    let(:resource) do
+      Puppet::Type.type(:oneview_storage_pool).new(
+        name: 'Storage Pool',
+        ensure: 'present',
+        data:
+            {
+              'name' => '172.18.8.11, PDU 1',
```
Comment (AsisBagga) [inline spec/unit/provider/oneview_storage_pool_spec.rb]: please remove the internal IP addresses.

File: spec/unit/provider/oneview_storage_pool_spec.rb [hunk 3]
```diff
@@ -19,33 +19,54 @@
...
+    let(:system_resource) do
+      Puppet::Type.type(:oneview_storage_system).new(
+        name: 'ThREEPAR-1',
+        ensure: 'present',
+        data:
+            {
+              'managedDomain' => 'TestDomain',
+              'credentials' =>
+              {
+                'ip_hostname' => '172.18.11.12',
```
Comment (AsisBagga) [inline spec/unit/provider/oneview_storage_pool_spec.rb]: same

**Review by AsisBagga — APPROVED** [review]
Before merging please make sure to remove the remove the internal IP Addresses as a part to follow the best practices.

File: examples/storage_pool.pp (hunk 1 above)
Comment (yuvirani) [reply examples/storage_pool.pp]: Reverted back

File: spec/unit/provider/oneview_storage_pool_spec.rb (hunk 2 above)
Comment (yuvirani) [reply spec/unit/provider/oneview_storage_pool_spec.rb]: Done

File: lib/puppet/provider/oneview_storage_pool/c7000.rb [hunk 4]
```diff
@@ -60,4 +67,10 @@ def setup_for_recreation
     storage_pool.first.remove
     true
   end
+
+  def set_storage_system
+    return unless @data['storageSystemUri'].empty?
+    storage_system_class = OneviewSDK.resource_named('StorageSystem', @client.api_version)
+    @data['storageSystemUri'] = storage_system_class.get_all(@client).first['uri']
```
Comment (VenkateshRavula) [inline lib/puppet/provider/oneview_storage_pool/c7000.rb]: we can have multiple storage systems added to oneview. So please let the user to provide the name or ip of storage system and you use "find_by" method using that name.
```
OneviewSDK.resource_named('StoragePool', api_version).find_by(@client, name: @data['storageSystem']).first['uri']
```

Comment (yuvirani) [reply lib/puppet/provider/oneview_storage_pool/c7000.rb]: Done.

File: lib/puppet/provider/oneview_storage_pool/c7000.rb [hunk 5]
```diff
@@ -37,12 +37,15 @@ def create
...
   def manage
+    return unless @client.api_version >= 500
     is_managed = @data.delete('isManaged')
     get_single_resource_instance.manage(is_managed)
     true
   end
 
   def reachable
+    return unless @client.api_version >= 500
+    @data['uri'] = OneviewSDK.resource_named('StoragePool', api_version).find_by(@client, name: @data['uri']).first['uri']
```
Comment (VenkateshRavula) [inline lib/puppet/provider/oneview_storage_pool/c7000.rb]: Give the decision to user whether to pass "name" or "uri". Keep the key names different.
```
temp_name = @data.delete('name')
@data['uri'] = OneviewSDK.resource_named('StoragePool', api_version).find_by(@client, name: temp_name).first['uri']
```

**Review by VenkateshRavula — APPROVED** [review]
LGTM

**Pull request merged**